CruCentralCoast/Community/CommunityGroups/CommunityGroupCell.swift

```diff
@@ -11,61 +11,81 @@ import UIKit
-        self.cellMask.layer.shadowRadius = 10
-        self.bannerImageView.layer.cornerRadius = 20
-        self.bannerImageView.layer.maskedCorners = [.layerMaxXMinYCorner, .layerMinXMinYCorner]
-        self.bannerImageView.layer.masksToBounds = true
     }
     
     func configure(with communityGroup: CommunityGroup) {
-        self.genderLabel.text = communityGroup.gender.rawValue.uppercased()
-        self.dayLabel.text = communityGroup.weekDay.rawValue.uppercased()
-        self.timeLabel.text = communityGroup.time
-        self.nameLabel.text = communityGroup.name
-        self.currentImageLink = communityGroup.imageLink
-        self.bannerImageView.image = nil
-        self.setImageVisibility(communityGroup.imageLink != nil && !communityGroup.imageLink!.isEmpty)
-        // Try to download the image, but only display it if this cell has not been reused
-        if let imageLink = communityGroup.imageLink {
-            ImageManager.instance.fetch(imageLink) { [weak self] image in
-                if let currentImageLink = self?.currentImageLink, currentImageLink == imageLink {
-                    DispatchQueue.main.async {
-                        self?.bannerImageView.image = image
-                    }
-                }
-            }
+        
+        
+        var leaderArray : [String] = []
+        for leader in communityGroup.leaders {
+            leaderArray.append(leader.name)
         }
-    }
-    
-    private func setImageVisibility(_ isVisible: Bool) {
-        if isVisible {
-            // If image size constraint doesn't exist, recreate it
-            if self.imageViewAspectRatioConstraint == nil {
-                self.imageViewAspectRatioConstraint = NSLayoutConstraint(item: self.bannerImageView, attribute: NSLayoutAttribute.width, relatedBy: NSLayoutRelation.equal, toItem: self.bannerImageView, attribute: NSLayoutAttribute.height, multiplier: 16/9, constant: 1)
-                self.imageViewAspectRatioConstraint.isActive = true
+        
+        if leaderArray.isEmpty {
+            if communityGroup.gender.rawValue.lowercased() == "male" {
+                self.nameLabel.text = "Men's Group"
+            }  else if communityGroup.gender.rawValue.lowercased() == "female" {
+                self.nameLabel.text = "Women's Group"
+            } else {
+                self.nameLabel.text = "Co-ed Group"
             }
         } else {
-            // If image size constraint exists, remove it
-            if self.imageViewAspectRatioConstraint != nil {
-                self.imageViewAspectRatioConstraint.isActive = false
-            }
+            self.nameLabel.text = leaderArray.joined(separator: ", ")
         }
+        
+        
+        
+        let gender = communityGroup.gender.rawValue.localizedCapitalized
+        let time = communityGroup.time ?? "N/A"
+        let year = communityGroup.year.rawValue.localizedCapitalized
+        
+        var captionArray = [gender,year,time]
```

Comment:
You could use `var captionArray = (time.isEmpty) ? [gender, year] : [gender, year, time]`

Reply:
nice, thanks